arwenmandos/util.go

```diff
@@ -198,3 +199,58 @@ func addESDTToVMInput(esdtData []*mj.ESDTTxData, vmInput *vmcommon.VMInput) {
 		}
 	}
 }
+
+func logGasTrace(ae *ArwenTestExecutor) {
+	length := len(ae.scenarioTraceGas)
+	if ae.scenarioTraceGas[length-1] {
+		runtime := ae.GetVMHost().Runtime()
+		scGasTrace := runtime.GetGasTrace()
+		arwen.SetLoggingForTests()
+		totalGasUsedInSmartContract := 0
+		for scAddress, gasTrace := range scGasTrace {
+			log.Trace("Gas Trace for: ", "SC Address", scAddress)
+			for functionName, value := range gasTrace {
+				totalGasUsed := uint64(0)
+				for _, usedGas := range value {
+					totalGasUsed += usedGas
+				}
+				log.Trace("GasTrace", "functionName:", functionName, ",  totalGasUsed:", totalGasUsed, ", numberOfCalls:", len(value))
+				totalGasUsedInSmartContract += int(totalGasUsed)
+			}
+			log.Trace("TotalGasUsedInSmartContract", "", totalGasUsedInSmartContract)
```

Comment:
Total gas used by the contract is `GasProvided` minus `GasRemaining` at the end. Instead, the value computed here is the total gas used by the contract _for API calls_. It's not all the gas.